### PR TITLE
Unity get connection and integration refactor

### PR DESCRIPTION
**Pull request by franklin-huang-mobiledgex**
1) Put the "Integration" files in their own folder in Scripts/. The directory layout is:

-Integration (Dir)
--- MobiledgeXIntegration.cs
--- PlatformIntegration (Dir)
----- PlatformIntegration.cs
----- CarrierInfoIntegration.cs
----- NetInterfaceIntegration.cs
----- UniqueIDIntegration.cs

Logic: 

-PlatformIntegration is a class that contains CarrierInfoClass, NetInterfaceClass, and UniqueIDClass objects as well as a NetworkInterfaceName object. 

-Those classes are implementations of the c# interfaces (CarrierInfo, NetInterface, UniqueID) with specific Unity handling per platform (IOS, Android, etc.) located in the other files. 

-MobiledgeXIntegration is the wrapper class that contains MatchingEngine and PlatformIntegration objects so that the all of the C# SDK functions and APIs can run in Unity (Might be a little confusing with this class -> too many wrappers). 

-Because the MatchingEngine constructor requires these as parameters in order for Unity to get device specific information, our code flow is more intuitive (Code inside MobiledgeXIntegration): 

```
PlatformIntegration integration = new PlatformIntegration();
MatchingEngine me = new MatchingEngine(integration.CarrierInfo, integration.NetInterface, integration.UniqueID);
```

Then we can access Device specific parameters (like unique ID, carrier name, local ip) like this:

`string uniqueID = me.UniqueID.GetUniqueID()`

2) Added GetConnection workflow to the PingPongGame app. The WebSocketClient object (ws) in the wsClient wrapper class is generated using the workflow.

(Note: Added optional path parameter for GetWebsocketConnection in the c# sdk, so that we can add path or query to the server uri)

### CODE REVIEW

**Comment by lgarner** [comment]
^ Whoops. Now have conflicts.

File: unity/PingPongGameExample/PingPongGame/Assets/Scripts/Integration/PlatformIntegration/CarrierInfoIntegration.cs [hunk 1]
```diff
@@ -0,0 +1,362 @@
...
+      }
+
+      AndroidJavaObject telManager = GetTelephonyManager();
+      if (telManager == null)
+      {
+        Debug.Log("Can't get telephony manager!");
+        return null;
+      }
+
+      if (!Permission.HasUserAuthorizedPermission(Permission.FineLocation))
```
Comment (lgarner) [inline unity/PingPongGameExample/PingPongGame/Assets/Scripts/Integration/PlatformIntegration/CarrierInfoIntegration.cs]: This seems to be READ_PHONE_STATE?

Comment (franklin-huang-mobiledgex) [reply unity/PingPongGameExample/PingPongGame/Assets/Scripts/Integration/PlatformIntegration/CarrierInfoIntegration.cs]: https://developer.android.com/reference/android/telephony/TelephonyManager.html#getAllCellInfo(). This only says Fine Location is required

Comment (lgarner) [reply unity/PingPongGameExample/PingPongGame/Assets/Scripts/Integration/PlatformIntegration/CarrierInfoIntegration.cs]: Great! Thanks for verifying.

**Pull request merged**